src/Data/Vec/Properties.agda

```diff
@@ -102,6 +109,15 @@ lookup⇒[]= (suc i) (_ ∷ xs) p    = there (lookup⇒[]= i xs p)
   []=⇒lookup∘lookup⇒[]= (x ∷ xs) (suc i) p    =
     []=⇒lookup∘lookup⇒[]= xs i p
 
+lookup-inject≤-take : ∀ m {n} (i : Fin m) (xs : Vec A (m + n)) →
+                      lookup xs (Fin.inject≤ i (m≤m+n m n)) ≡ lookup (take m xs) i
```

Comment:
Probably best to extract the proof `m ≤ m + n` to its own variable and not assume that the proof was created via the `m≤m+n` lemma. Otherwise if someone proves the inequality a different way it's a bit of a pain for them to apply the lemma. You could still have this form as a special case with a `\prime` on the end if you wanted to this in for convenience.